daemon/daemon_unix.go

```diff
@@ -834,24 +834,15 @@ func overlaySupportsSelinux() (bool, error) {
 	}
 	defer f.Close()
 
-	var symAddr, symType, symName, text string
-
 	s := bufio.NewScanner(f)
 	for s.Scan() {
-		if err := s.Err(); err != nil {
-			return false, err
-		}
-
-		text = s.Text()
-		if _, err := fmt.Sscanf(text, "%s %s %s", &symAddr, &symType, &symName); err != nil {
-			return false, fmt.Errorf("Scanning '%s' failed: %s", text, err)
-		}
-
-		// Check for presence of symbol security_inode_copy_up.
-		if symName == "security_inode_copy_up" {
+		if strings.HasSuffix(s.Text(), " security_inode_copy_up") {
```

Comment:
So we need a `text := strings.TrimSpace(s.Text())` here as well?

Reply:
Nope, because the field we are looking for is the last one, and there should be no whitespace after it, just a newline.

In other words, this code equals `grep ' security_inode_copy_up$' /proc/kallsyms`

Reply:
So there's never anything after `security_inode_copy_up`? Is this future proof?

Reply:
> So there's never anything after security_inode_copy_up? Is this future proof?

This file has been added in linux kernel 2.5.71, which will turn 17 in about 2 months. It has been that way since. I doubt they will break it.